src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserManagementService.cs

```diff
@@ -247,45 +290,59 @@ await _usersAndRolesDbContext.UserInvites
                 });
         }
 
-        public async Task<bool> UpdateAsync(string userId, string roleId)
+        public async Task<Either<ActionResult, Unit>> UpdateUser(string userId, string roleId)
         {
-            var user = await _usersAndRolesDbContext.Users.FirstOrDefaultAsync(i => i.Id == userId);
-            var userRole = await _usersAndRolesDbContext.UserRoles.FirstOrDefaultAsync(i => i.UserId == userId);
+            return await _userService
+                .CheckCanManageAllUsers()
+                .OnSuccess<ActionResult, Unit, Unit>(async () =>
+                {
+                    var user = await _usersAndRolesDbContext.Users.FirstOrDefaultAsync(i => i.Id == userId);
+                    if (user == null)
+                    {
+                        return ValidationActionResult(UserDoesNotExist);
+                    }
 
-            if (user == null || userRole == null)
-            {
-                return false;
-            }
+                    var userRole = await _usersAndRolesDbContext.UserRoles.FirstOrDefaultAsync(i => i.UserId == userId);
+                    if (userRole == null)
+                    {
+                        return ValidationActionResult(InvalidUserRole);
+                    }
 
-            await _userManager.RemoveFromRoleAsync(user, GetRoleName(userRole.RoleId));
-            await _userManager.AddToRoleAsync(user, GetRoleName(roleId));
+                    await _userManager.RemoveFromRoleAsync(user, GetRoleName(userRole.RoleId));
+                    await _userManager.AddToRoleAsync(user, GetRoleName(roleId));
 
-            return true;
+                    return Unit.Instance;
+                });
         }
 
-        public List<UserReleaseRoleViewModel> GetUserReleaseRoles(string userId)
+        public async Task<Either<ActionResult, List<UserReleaseRoleViewModel>>> GetUserReleaseRoles(string userId)
```

Comment:
It seems like this method should only need to be `private` and only called when getting the User. The front end is calling this separately but the result is already included in the `User` that it's already fetched, so with a little front end refactoring the public `GetUserReleaseRoles` could be scrapped.

Reply:
Made a ticket to address this issue separately: https://dfedigital.atlassian.net/browse/EES-1787